SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+            $NewAppVolume = Read-Host "Your AppVolume: "
+            If ([string]::IsNullOrEmpty($NewAppVolume)) {
+                Write-Message -Level Verbose -Message "AppVolume remains on " $AppVolume
+            }
+            else {
+                $NewAppVolume = $NewAppVolume -replace "\\$"
+                $NewAppVolume = $NewAppVolume -replace ":$"
+                $AppVolume = $NewAppVolume
+                Write-Message -Level Verbose -Message "AppVolume moved to " $AppVolume
+            }
+
+            Write-Message -Level Verbose -Message "BackupVolume: " $BackupVolume
+            $NewBackupVolume = Read-Host "Your BackupVolume: "
+            If ([string]::IsNullOrEmpty($NewBackupVolume)) {
+                Write-Message -Level Verbose -Message "BackupVolume remains on " $BackupVolume
+            }
+            else {
+                $NewBackupVolume = $NewBackupVolume -replace "\\$"
+                $NewBackupVolume = $NewBackupVolume -replace ":$"
+                $BackupVolume = $NewBackupVolume
+                Write-Message -Level Verbose -Message "BackupVolume moved to " $BackupVolume
+            }
+        }
+        Default {Write-Message -Level Verbose -Message "Drives agreed, continuing"; }
+    }
+
+
+    # Out-File -FilePath C:\Temp\ConfigurationFile2.ini -InputObject $startScript
+
+    (Get-Content -Path $configini).Replace('SQLBACKUPDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"', 'SQLBACKUPDIR="' + $BackupVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLUSERDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLUSERDBDIR="' + $DataVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $configini
+
+    (Get-Content -Path$configini).Replace('SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLTEMPDBDIR="' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"', 'SQLUSERDBLOGDIR="' + $LogVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLSYSADMINACCOUNTS="WIN-NAJQHOBU8QD\Administrator"', 'SQLSYSADMINACCOUNTS="' + $SqlServerAccount)| Out-File $configini
+
+    If ($SaveFile -eq "Yes") {
+        $SaveFileLocation = Read-Host "Please enter your preferred directory for saving a copy of the configuration file: "
+        Copy-Item $configini -Destination $SaveFileLocation
+    }
+
+    IF ($Authentication -eq "Mixed Mode") {
+        $SAPassW = [PsCredential](Get-Credential -UserName "SA"  -Message "Please Enter the SA Password.")
```

Comment:
I'd do this check closer to the top and use that `Credential` parameter. Unless that is being used for remote stuff, then do like `[pscredential]$SaCredential` 

so that we can reduce interruption to automation as much as possible